test/Test/Tinfoil/AEAD/AESGCM/Data.hs

```diff
@@ -0,0 +1,32 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE TemplateHaskell #-}
+module Test.Tinfoil.AEAD.AESGCM.Data where
+
+import           Data.Bits (shiftR, (.&.))
+import           Data.Word (Word64)
+
+import           P
+
+import           System.IO
+
+import           Test.QuickCheck hiding ((.&.))
+import           Test.Tinfoil.Arbitrary ()
+
+import           Tinfoil.AEAD.AESGCM.Data
+
+unpackInvocationField :: Word64 -> InvocationField
+unpackInvocationField w =
+  let
+    rf = RandomField . fromIntegral $ w `shiftR` 32
+    ic = InvocationCount . fromIntegral $ w .&. 0x00000000ffffffff
+  in
```

Comment:
Yeah, this is just me being lazy (in a paranoid way) about the possibility of overflow. You're right, I will rework this to be more efficient.